datadog_checks_dev/datadog_checks/dev/tooling/commands/dep.py

```diff
@@ -108,3 +117,162 @@ def freeze():
     lines = sorted(set(f'{d[1]}\n' for d in data))
 
     write_file_lines(static_file, lines)
+
+
+@dep.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help="Update integrations' dependencies so that they match the Agent's static environment",
+)
+def sync():
+    all_agent_dependencies, errors = read_agent_dependencies()
+
+    if errors:
+        for error in errors:
+            echo_failure(error)
+        abort()
+
+    files_updated = 0
+    checks = sorted(get_valid_checks())
+    for check_name in checks:
+        check_dependencies, check_errors = read_check_dependencies(check=check_name)
+
+        if check_errors:
+            for error in check_errors:
+                echo_failure(error)
+            abort()
+
+        deps_to_update = {
+            check_dependency_definition: agent_dep_version
+            for check_dependency_name, check_dependency_definitions in check_dependencies.items()
+            for version, dependency_definitions in check_dependency_definitions.items()
+            for check_dependency_definition in dependency_definitions
+            for agent_dep_version, agent_dependency_definitions in all_agent_dependencies[check_dependency_name].items()
+            for agent_dependency_definition in agent_dependency_definitions
+            # look for the dependency with the same marker and name since the version can be different
+            if check_dependency_definition.same_name_marker(agent_dependency_definition)
+            and version != agent_dep_version
+        }
+
+        if deps_to_update:
+            files_updated += 1
+            check_req_file = get_check_req_file(check_name)
+            old_lines = read_file_lines(check_req_file)
+            new_lines = old_lines.copy()
+
+            for dependency_definition, new_version in deps_to_update.items():
+                dependency_definition.requirement.specifier = new_version
+                new_lines[dependency_definition.line_number] = f'{dependency_definition.requirement}\n'
+
+            write_file_lines(check_req_file, new_lines)
+
+    if not files_updated:
+        echo_info('All dependencies synced.')
+
+    echo_info(f'Files updated: {files_updated}')
+
+
+async def get_version_data(url):
```

Comment:
A lot of the fetching logic for `ddev dep updates` is borrowed from `datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/licenses.py` as they both involve scraping pypi; I could abstract this out in this PR or a separate one